content/teknologi/altinnstudio/altinn-api/app-api/instances/_index.md

```diff
@@ -0,0 +1,148 @@
+The name of the parts must correspond to element types defined in the application metadata. 
+
+```http
+POST {basePath}
+```
+
+![Flow chart for instantiation](instantiate-for-an-instance-owner.png "Instantiate for instance owner")
+
+A multipart formdata should contain the instance json document and the data element files of the instance.
+The first part should be *instance* which contains the json template to create an instance from.
+
+The subsequent parts must have a name that correspond to the element types defined in application metadata.
+They may have a filename. Hence the *model1* and *certificate* names correspond to data types defined in the application metadata.
+If more data elements are needed they must be defined in the application metadata.
+
+```http {linenos=false,hl_lines=[5,10,15]}
+Content-Type: multipart/form-data; boundary="abcdefg"
+
+--abcdefg
+Content-Type: application/json; charset=utf-8
+Content-Disposition: form-data; name="instance"
+{ ... }
+
+--abcdefg
+Content-Type: application/xml
+Content-Disposition: form-data; name="model1"
+<xml> ... </xml>
+
+--abcdefg
+Content-Type: application/pdf
+Content-Disposition: form-data; name="certificate"; filename=certificate.pdf
+%PDF-1.4
+%Óëéá
+1 0 obj
+...
+
+--abcdefg--
+```
+
+This call will return the instance metadata document that was created. 
+
+## Complete instance
+
+Endpoint used by Application Owner to mark an instance as completed. Technically this only means that the instance is no longer needed by the Application Owner. Altinn can delete the instance and associated data should the Application user decide to do so.
+
+```http
+POST {basePath}/{instanceOwnerPartyId}/{instanceGuid}/complete
+```
+
+The request does not use the request body for anything even though it is a POST request.
+
+## Update sub status
+
+The instance [sub status](https://altinn.github.io/docs/altinn-studio/app-creation/instance/#substatus) is used to give an end user further details about the state of their instance. Currently, only application owner is allowed to update substatus for an instance. Include the new substatus in the body of the requests as a json.
+
+```http
+PUT {basePath}/{instanceOwnerPartyId}/{instanceGuid}/substatus
+```
+
+**Request body**
+```json
+{
+    "label": "some.label",
+    "description": "Beskrivelse i klarteskst"
+}
+```
+The values can be referencing text resource names from the language files or be regular text.
+
+
+## Delete instance
+
+Endpoint for marking an instance as deleted. This can be used by both Application Owner and User to delete an instance. The endpoint has an optional parameter called **hard** that can be used to indicate the type of delete that is wanted. Hard delete means that the instance will be unrecoverable and deleted completely by Altinn within a few days. Setting the parameter to false or simply omitting it, will move the instance to a recycle bin. 
```

Comment:
Deleted within a few days is a truth with modifications I guess.. Any way to communicate that there are additional factors that affect the actual deletion time?

Reply:
Or are we assuming that only drafts are deleted using this endpoint? Befause in that case, the statement would be accurate.